src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -1555,239 +1755,47 @@ private void EndTagBlock(Stack<Tuple<SyntaxToken, SourceLocation>> tags, bool co
-
-        private bool CurrentScriptTagExpectsHtml()
-        {
-            var blockBuilder = Context.Builder.CurrentBlock;
-
-            Debug.Assert(blockBuilder != null);
-
-            var typeAttribute = blockBuilder.Children
-                .OfType<Block>()
-                .Where(block =>
-                    block.ChunkGenerator is AttributeBlockChunkGenerator &&
-                    block.Children.Count() >= 2)
-                .FirstOrDefault(IsTypeAttribute);
-
-            if (typeAttribute != null)
-            {
-                var contentValues = typeAttribute.Children
-                    .OfType<Span>()
-                    .Where(childSpan => childSpan.ChunkGenerator is LiteralAttributeChunkGenerator)
-                    .Select(childSpan => childSpan.Content);
-
-                var scriptType = string.Concat(contentValues).Trim();
-
-                // Does not allow charset parameter (or any other parameters).
-                return string.Equals(scriptType, "text/html", StringComparison.OrdinalIgnoreCase);
-            }
-
-            return false;
-        }
-
-        private static bool IsTypeAttribute(Block block)
-        {
-
-            if (!(block.Children.First() is Span span))
-            {
-                return false;
-            }
-
-            var trimmedStartContent = span.Content.TrimStart();
-            if (trimmedStartContent.StartsWith("type", StringComparison.OrdinalIgnoreCase) &&
-                (trimmedStartContent.Length == 4 ||
-                ValidAfterTypeAttributeNameCharacters.Contains(trimmedStartContent[4])))
-            {
-                return true;
-            }
-
-            return false;
-        }
-
-        public void ParseRazorBlock(Tuple<string, string> nestingSequences, bool caseSensitive)
-        {
-            if (Context == null)
-            {
-                throw new InvalidOperationException(Resources.Parser_Context_Not_Set);
-            }
-
-            using (PushSpanConfig(DefaultMarkupSpan))
-            {
-                Span.Start = CurrentLocation;
-
-                using (Context.Builder.StartBlock(BlockKindInternal.Markup))
-                {
-                    NextToken();
-                    CaseSensitive = caseSensitive;
-                    if (nestingSequences.Item1 == null)
-                    {
-                        NonNestingSection(nestingSequences.Item2.Split());
-                    }
-                    else
-                    {
-                        NestingSection(nestingSequences);
-                    }
-                    AddMarkerTokenIfNecessary();
-                    Output(SpanKindInternal.Markup);
-                }
+                return SyntaxFactory.MarkupBlock(builder.ToList());
             }
         }
 
-        private void NonNestingSection(string[] nestingSequenceComponents)
+        private void NonNestingSection(in SyntaxListBuilder<RazorSyntaxNode> builder, string[] nestingSequenceComponents)
```

Comment:
The word `section` here is a bit confusing. What does this guy do exactly? Is it the bits that parse Html tags in `@{...}` to ensure they match? A bit confused is all

Reply:
I've removed this and cleaned up a little. We parse section blocks using this logic and I assume that's is where the name came from. Anyway, we don't call into `NonNestingSection` anymore and I've renamed `NestingSection` to `NestingBlock`. 
What this does is handle nesting of blocks inside Razor Blocks. E.g,

```HTML
@section MySection {
<div>hello</div>
@if (true) {
  // I am a nested block inside a Razor block so keep track of all the open curlys (aka. nesting sequences)
}
}
```

Reply:
Correct me if I am wrong though. I could be misunderstanding some legacy piece.